client/editor/editor.go

```diff
@@ -90,25 +118,42 @@ func (e *Editor) AddRune(r rune) {
 	} else {
 		e.Text = append(e.Text[:e.Cursor], r)
 	}
-	e.Cursor++
+	e.MoveCursor(1, 0)
 }
 
 // Draw updates the UI by setting cells with the editor's content.
 func (e *Editor) Draw() {
 	_ = termbox.Clear(termbox.ColorDefault, termbox.ColorDefault)
 	cx, cy := e.calcCursorXY(e.Cursor)
+
+	// draw cursor x position relative to row offset
+	if cx-e.GetColOff() > 0 {
+		cx -= e.GetColOff()
+	}
+
+	// draw cursor y position relative to row offset
+	if cy-e.GetRowOff() > 0 {
+		cy -= e.GetRowOff()
+	}
+
 	termbox.SetCursor(cx-1, cy-1)
+	// find the starting and ending row of the termbox window.
+	yStart := e.GetRowOff()
+	yEnd := yStart + e.GetHeight() - 1 // -1 accounts for the status bar
+
+	// find the starting ending column of the termbox window.
+	xStart := e.GetColOff()
```

Comment:
There's something wrong with the current implementation: I tried adding a long piece of text (~90 characters; which is greater than the average width of a small terminal window (80 characters)); the scroll works nicely, however there's an issue with the final pieces of text getting lost. I don't think that the CRDT has lost content; as this PR involves changes to the UI only.

This is a weird, unusual issue that I was unable to debug, could you please take a look at it? It could be reproduced easily by adding a line with content exceeding the width of the terminal.

Reply:
For example, if I added `really loooo......ooong text`, the first scroll works as intended, but subsequent scrolls lose some content. (`really loooo......ooong`; `text` is not visible now)

Reply:
I'll take a look! Are you copying and pasting or typing it all out?

Reply:
Typing it all out.

Reply:
I'm having trouble reproducing the losing text issue (though I found a couple others).

I'm typing 50+ characters past the edge of the terminal window, all in one line, scrolling all the way back to the beginning of the line with the arrow key, then scrolling all the way back to the end of the line, but all the text is still there for me.

Reply:
![download](https://user-images.githubusercontent.com/53528139/232233906-d288d73a-61d0-4e3c-afc2-9c7980f695d3.gif)

Sorry for the delay, the GIF generation took some time. Here's what I did.

Reply:
Alright, I pushed a change and a couple more tests. Hopefully it fixes the strange cursor behavior when moving between lines.

Reply:
Looks good now!